### ISSUE TITLE
A context-related bug

### ISSUE THREAD

**Issue by NWU-NISL** [mozilla/rhino#684]
When executing the two testcases with the same function, testcase2 only inserts an irrelevant code segment on the basis of testcase1, but the execution results of the two test cases are different. Therefore, I think it is a bug.

### Command:
``` shell
java -jar rhino/rhino-1.7.12.jar -debug -version 200 testcase.js
```

### testcase1:
``` javascript
var NISLFuzzingFunc = function (a, nislMutationParameter0) {
   //var v1 = a.replace(/(.)/g, function (a, b) {
   //    return b ? a : '';
   //});
   var v1 = a;
   var v2 = v1.replace(nislMutationParameter0, function (b, c) {
        print(c);
    }); 
};
var NISLParameter0 = 'hello world';
var nislMutationArgument0 = []; 
var NISLCallingResult = NISLFuzzingFunc(NISLParameter0, nislMutationArgument0);
```

### output  1:
```
0
```

### testcase2:
``` javascript
var NISLFuzzingFunc = function (a, nislMutationParameter0) {
   var v1 = a.replace(/(.)/g, function (a, b) {
       return b ? a : '';
   });
   var v1 = a;
   var v2 = v1.replace(nislMutationParameter0, function (b, c) {
        print(c);
    }); 
};
var NISLParameter0 = 'hello world';
var nislMutationArgument0 = []; 
var NISLCallingResult = NISLFuzzingFunc(NISLParameter0, nislMutationArgument0);
```
###output  2:
```
d
```

**Comment by rbri** [comment]
Wow, cool catch.

**Comment by rbri** [comment]
Fixed with commit ef02aa3992c4b834b9cd95a29d0027893fa872fb. Please close....
